### ISSUE TITLE
Primitive but only for JSON

### ISSUE THREAD

**Issue by kitsoRik** [krzkaczor/ts-essentials#266]
Library contains good type `Primitive`, its very useful, it's good idea to have the same type but for types from JSON, something like that: `type JSONValue = string | number | boolean | null;`

**Comment by Beraliv** [comment]
@kitsoRik thank you again for the good suggestion!

At the moment I'm preparing for the conference and will be back next week!

For me the plan sounds good, I really like the idea and most probably we will start adding it soon

cc @krzkaczor

**Comment by kitsoRik** [comment]
@Beraliv Yeah, this is a good package that I install in all projects where I use TypeScript, and I would like to have there all common types. That's not a problem for me (about your busy status). Anyway, I need this type in my current project and I just put it in my types folder.


FYI, now it looks like this:

```typescript
export type JsonFieldType =
  | string
  | number
  | boolean
  | null
  | { [path: string]: JsonFieldType }
  | JsonFieldType[];
```

I'm not sure this is finished implementation, so you need to test all cases.

**Comment by kitsoRik** [comment]
Hi. Regarding my implementation above, found a strange behaviour with inherited classes. 
```typescript
export type JsonFieldType =
  | string
  | number
  | boolean
//   | undefined // uncomment to fix
  | null
  | { [path: string]: JsonFieldType }
  | JsonFieldType[];

abstract class A {
    method(): JsonFieldType {
        if(Math.random() > 0.5) {
            return { q: 1 };
        }

        return { q: 1 };
    }
}

class B extends A {
    method() {
        if(Math.random() > 0.5) {
            return { q: 1 };
        }

        return { w: 1 };
    }
}
```

**Comment by Beraliv** [comment]
Hmm, that's weird

I tried it here – https://tsplay.dev/WPx85W and doesn't have any issues

What TypeScript do you have?

**Comment by kitsoRik** [comment]
> Hmm, that's weird
> 
> I tried it here – https://tsplay.dev/WPx85W and doesn't have any issues
> 
> What TypeScript do you have?

Yeah, because you uncommented some strings. It works fine without `undefined` (only with implicitly defined return type of method)

**Comment by Beraliv** [comment]
> > Hmm, that's weird
> > I tried it here – https://tsplay.dev/WPx85W and doesn't have any issues
> > What TypeScript do you have?
> 
> Yeah, because you uncommented some strings. It works fine without `undefined` (only with implicitly defined return type of method)

Ah, sorry, I needed to have it without uncommenting

**Comment by Beraliv** [comment]
Also this can be connected with the future implementation of `JsonFieldType` – https://github.com/krzkaczor/ts-essentials/issues/276

**Comment by Beraliv** [comment]
Also found interesting implementation here – https://github.com/Microsoft/TypeScript/issues/1897#issuecomment-710744173

**Comment by Beraliv** [comment]
Hey @rpidburachynskyi! I'm planning to include it in the next minor change, would you be interested in having this type now?

**Comment by Beraliv** [comment]
This will be released as part of https://github.com/ts-essentials/ts-essentials/pull/438 (v10.2.0)

**Comment by Beraliv** [comment]
Released today under v10.2.0
